src/node/net.ts

```diff
@@ -1110,10 +1110,23 @@ async function startRead(socket: SocketClass): Promise<void> {
   const reader = socket._handle.reader;
   try {
     while (socket._handle.reading === true) {
+      const generatedBuffer = socket[kBufferGen]?.();
+
+      // Let's be extra cautious here and handle nullish values.
+      if (generatedBuffer == null || generatedBuffer.length === 0) {
+        // When reading a static buffer with fixed length, it's highly likely to
+        // read the whole buffer in a single take, which will make the second
+        // operation to read an empty buffer.
+        //
+        // Workerd throws the following exception when reading empty buffers
```

Comment:
fwiw, throwing here is mandated by the spec. It's not something specific to our implementation as this comment may suggest.

Reply:
Agreed. AFAICT, node.js uses nodejs streams which doesn't require throwing an error, whereas workerd uses web streams in here.